src/lib/device/virtual_dev.cpp

```diff
@@ -262,26 +252,26 @@ BlkAllocStatus VirtualDev::alloc_blks_from_chunk(blk_count_t nblks, blk_alloc_hi
     return status;
 }
 
-/*bool VirtualDev::free_on_realtime(BlkId const& b) {
-    Chunk* chunk = m_dmgr.get_chunk_mutable(b.chunk_num());
-    return chunk->blk_allocator_mutable()->free_on_realtime(b);
-}*/
+void VirtualDev::free_blk(BlkId const& bid, VDevCPContext* vctx) {
+    auto do_free_action = [this](auto const& b, VDevCPContext* vctx) {
+        if (vctx && (m_allocator_type != blk_allocator_type_t::append)) {
+            // We don't want to accumulate here for append blk allocator.
+            vctx->m_free_blkid_list.push_back(b);
+        } else {
+            BlkAllocator* allocator = m_dmgr.get_chunk_mutable(b.chunk_num())->blk_allocator_mutable();
+            if (m_auto_recovery) { allocator->free_on_disk(b); }
```

Comment:
nit: is this needed since it does nothing.

Reply:
It does free_on_disk on cases where we use persistent blkallocator, but not part of CP. So this code is needed.

Reply:
I see. It also fall through to non-append blkalloctor with vctx equal to nullptr;